### PR TITLE
Added IP test plugins

### PR DESCRIPTION
**Pull request by priyamsahoo**
##### SUMMARY

This PR adds 24 IP test plugins along with their unit tests

##### ISSUE TYPE

- New Plugins Pull Request

##### COMPONENT NAME

1. in_any_network
2. in_network
3. in_one_network
4. ip_address
5. ip
6. ipv4
7. ipv4_address
8. ipv4_hostmask
9. ipv4_netmask
10. ipv6
11. ipv6_address
12. ipv6_ipv4_mapped
13. ipv6_sixtofour
14. ipv6_teredo
15. loopback
16. mac
17. multicast
18. private
19. public
20. reserved
21. subnet_of
22. supernet_of
23. unspecified
24. resolvable
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,36 +21,55 @@ PEP440 is the schema used to describe the versions of Ansible.
...
-[ansible.utils.to_paths](https://github.com/ansible-collections/ansible.utils/blob/main/docs/ansible.utils.to_paths_filter.rst)|Flatten a complex object into a dictionary of paths and values
-[ansible.utils.validate](https://github.com/ansible-collections/ansible.utils/blob/main/docs/ansible.utils.validate_filter.rst)|Validate data with provided criteria
+[ansible.utils.get_path](https://github.com/ansible-collections/ansible.utils/blob/network-test-plugins/docs/ansible.utils.get_path_filter.rst)|Retrieve the value in a variable using a path
+[ansible.utils.index_of](https://github.com/ansible-collections/ansible.utils/blob/network-test-plugins/docs/ansible.utils.index_of_filter.rst)|Find the indices of items in a list matching some criteria
+[ansible.utils.to_paths](https://github.com/ansible-collections/ansible.utils/blob/network-test-plugins/docs/ansible.utils.to_paths_filter.rst)|Flatten a complex object into a dictionary of paths and values
+[ansible.utils.validate](https://github.com/ansible-collections/ansible.utils/blob/network-test-plugins/docs/ansible.utils.validate_filter.rst)|Validate data with provided criteria
```
Comment (Qalthos) [inline README.md:27]: The branch should not have changed here

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (priyamsahoo) [reply README.md]: This is resolved in eb13cfb.

**Comment by ashwini-mhatre** [comment]
@priyamsahoo check pipeline is failing can you please fix the issues?

**Review by ganeshrn — COMMENTED** [review]
Can you please split the single big PR into multiple PR's so that it is bit easier to review and tack changes. Also I can see some files  completely commented out for eg: `plugins/test/general.py`. If it is not required those files can be removed

**Comment by ashwini-mhatre** [comment]
The code changes are already merged so closing this

**Pull request closed without merging**